src/Symfony/Bundle/SecurityBundle/Tests/Functional/SecurityTest.php

```diff
@@ -31,4 +33,44 @@ public function testServiceIsFunctional()
         $this->assertTrue($security->isGranted('ROLE_USER'));
         $this->assertSame($token, $security->getToken());
     }
+
+    public function userWillBeMarkedAsChangedIfRolesHasChangedProvider()
+    {
+        return [
+            [true],
+            [false],
+        ];
+    }
+
+    /**
+     * @dataProvider userWillBeMarkedAsChangedIfRolesHasChangedProvider
+     */
+    public function testUserWillBeMarkedAsChangedIfRolesHasChanged($roleAsObject)
```

Comment:
Please add a bool type hint and also the variable name is a bit confusing to me 🤔

Reply:
Maybe instead of providing true false, provide the role string and/or role object via the provider as $admin and $user to the test and don’t use a type hint as explained in my previous comment

Reply:
1/2 @nicolas-grekas is right 3.4 targets PHP 5.5
2/2 I'll remove a flag in favor of providing roles as string/or role object